packages/cdktf/lib/terraform-local.ts

```diff
@@ -0,0 +1,49 @@
+import { Construct } from "constructs";
+import { TerraformElement } from "./terraform-element";
+import { Token } from "./tokens";
+
+export class TerraformLocal extends TerraformElement {
+    private _expression: any;
+
+    constructor(scope: Construct, id: string, expression: any) {
+        super(scope, id);
+
+        this._expression = expression;
+    }
+
```

Comment:
I think it would make sense to add the following:

## Typescript 

```ts
    public toString(): string {
        return this.asString
    }
```

This would allow to use it like `my-first-table-${local}` rather than `my-first-table-${local.asString}` in Typescript. 

## Python

I did try it for Python but it doesn't work: `"my-first-sns-topic-<cdktf.TerraformLocal object at 0x10cd45190>"`

Defining the required method for Python in the Typescript class only works as `@internal` and therefore wouldn't be compiled via jsii.

```ts
 /** @internal */
    public __str__(): string {
        return this.asString
    }
```

There's an issue somehow related to this at Jsii: https://github.com/aws/jsii/issues/380

Other ideas?

Reply:
I agree it would be nice to directly reference the local value. I don't really think `toString` is the answer since I would expect users to have already set the local to the value they desire and `${local}` seems about the same as `local.asString` to me.

We could make properties accept and `IResolvable` interface in addition to their normal type, but I think we'll run into problems with some other jsii languages.

Reply:
Ok, let's leave it like this for now and come back to this later 👍